docs/data/horizon/api-reference/errors/error-handling.mdx

```diff
@@ -0,0 +1,214 @@
+2. Transaction submissions (a `POST /transactions`, `POST /transactions_async`).
+
+There are many possible error codes when executing these actions, and you can typically handle these error codes using the following strategies:
+
+- Request adjustments: adjusting the request to resolve structural errors with queries or transaction submissions. Suppose you’ve included a bad parameter, malformed your XDR, or otherwise didn’t follow the endpoint’s specification. In these cases, resolve the error by referencing the details or result codes of the error response.
+- Polling and retrying: this is the recommended way to work around latency or congestion issues encountered along the pipeline between your computer and the Stellar network, which can sometimes happen due to the nature of distributed systems.
+
+## Error Handling for Queries
+
+Many `GET` requests have specific parameter requirements, and while the SDKs can help enforce them, you can still pass invalid arguments (for example, an asset string that isn’t SEP-11 compatible) that error out every time. In this scenario, there’s nothing you can do aside from following the API specification. The `extras` field of the error response will often clue you in on where to look and what to look for.
+
+<CodeExample>
+
+```bash
+curl -s https://horizon-testnet.stellar.org/claimable_balances/0000 | jq '.extras'
+{
+    "invalid_field": "id",
+    "reason": "Invalid claimable balance ID"
+}
+```
+
+</CodeExample>
+
+Note that the SDKs make it a point to distinguish an invalid request (as above) versus a missing resource (a `404 Not Found`) (for example, the generic `NetworkError` versus a `NotFoundError` in the JavaScript SDK), where the latter might not be considered an error depending on your situation.
+
+## Error Handling for Transaction Submissions
+
+Horizon currently supports two types of transaction submission endpoints:
+
+1. `/transactions_async`: Horizon submits a transaction to Stellar-Core in an asynchronous manner and passes the relevant Stellar-Core response back immediately to the client. It is then the client's responsibility to poll for the status of that transaction.
+2. `/transactions`: Horizon submits a transaction to Stellar-Core and then waits for it to be ingested in the database. It will either timeout while waiting or return a transaction response on success.
+
+There are some resolution strategies that are common between the 2 endpoints while some strategies are more endpoint specific.
+
+### Request Adjustments
+
+Certain transaction submission failures also need adjustments to succeed.
+
+- If the XDR is malformed, or the transaction is otherwise invalid, you’ll encounter a `400 Bad Request` (for example, see excluding a source account). Both transactions and their operations can be easily malformed: look at the `extras.result_codes` field for details and cross-reference them with the appropriate result codes documentation to determine specifics.
```

Comment:
Is this supposed to be a cross-reference? If not, `see` doesn't make sense here:

```suggestion
- If the XDR is malformed, or the transaction is otherwise invalid, you’ll encounter a `400 Bad Request` (for example, an invalid source account). Both transactions and their operations can be easily malformed or invalid: look at the `extras.result_codes` field for details and cross-reference them with the appropriate result codes documentation to determine specifics.
```

Alternatively, you could say `(for example, [BAD_SEQ](#managing-specific-errors) for an invalid source account)`.

Reply:
(Also, malformed tells me "bad encoding" vs. invalid which is "valid encoding but not a valid tx in that moment".)